### PR TITLE
fix: show original in new tab for local storage #795

### PR DESCRIPTION
**Pull request by FrolYo**

### CODE REVIEW

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
 using Stl.IO;
 
 namespace ActualChat.Blobs.Internal;
 
 internal class LocalFolderBlobStorage : IBlobStorage
 {
-    private readonly FilePath _directory;
+    private IContentTypeProvider? _contentTypeProvider;
+    private FilePath Directory { get; }
+    private IServiceProvider ServiceProvider { get; }
```
Comment (alexyakunin) [inline src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs:11]: Let's be consistent & call it Services.

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+using Microsoft.AspNetCore.StaticFiles;
 using Stl.IO;
 
 namespace ActualChat.Blobs.Internal;
 
 internal class LocalFolderBlobStorage : IBlobStorage
 {
-    private readonly FilePath _directory;
+    private IContentTypeProvider? _contentTypeProvider;
+    private FilePath Directory { get; }
```
Comment (alexyakunin) [inline src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs:10]: The agreement is:
- Resolved services are always props
- Everything else is usual (i.e. can be a private field, property, etc. -whatever is reasonable).

And here it's reversed: _contentTypeProvider must be a property, Directory must be a field.

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs [hunk 3]
```diff
@@ -50,8 +56,8 @@ public async Task Write(string path, Stream dataStream, string contentType, Canc
 
         ValidatePath(path);
 
-        var fullPath = _directory & path;
-        Directory.CreateDirectory(fullPath.DirectoryPath);
+        var fullPath = Directory & path;
+        System.IO.Directory.CreateDirectory(fullPath.DirectoryPath);
```
Comment (alexyakunin) [inline src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs:60]: Maybe makes sense to use another name in such cases - e.g. RootDirectory, etc.

**Review by alexyakunin — CHANGES_REQUESTED** [review]

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs (hunk 1 above)
Comment (FrolYo) [reply src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs]: done

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs (hunk 2 above)
Comment (FrolYo) [reply src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs]: [discussed] now it's good

File: src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs (hunk 3 above)
Comment (FrolYo) [reply src/dotnet/Core/Blobs/Internal/LocalFolderBlobStorage.cs]: done

**Pull request merged**